e2e/browser/pages/AppPage.ts

```diff
@@ -0,0 +1,62 @@
+import { expect, Locator, Page } from "@playwright/test";
+
+type AppPageOptions = {
+  clientApplicationUrl: string;
+  fetchTimeout: number;
+};
+
+export class AppPage {
+  readonly page: Page;
+  private readonly url: string;
+  private readonly fetchTimeout: number;
+  private readonly fetchResponseText: Locator;
+
+  constructor(page: Page, options: AppPageOptions) {
+    this.page = page;
+    this.url = options.clientApplicationUrl;
+    this.fetchTimeout = options.fetchTimeout;
+
+    this.fetchResponseText = this.page.locator(
+      '[data-testid="fetch_response_textbox"]'
+    );
+  }
+
+  async start() {
+    await this.page.goto(this.url);
+  }
+
+  async waitForReady() {
+    return this.page.waitForURL(this.url);
+  }
+
+  // async disableReauthorize() {
+  //   return this.page.uncheck('[data-testid="login_reauthorize"]');
+  // }
+
+  async startLogin(url: string) {
+    await this.page.fill('[data-testid="idp_input"]', url);
+
+    // Click the login button:
+    await this.page.click('[data-testid="idp_login_button"]');
+    await this.page.waitForNavigation({
+      waitUntil: "networkidle",
+    });
+  }
+
+  async fetchResource(url: string) {
```

Comment:
Might be good to move both `fetchResource` and `getFetchResponse` and related code into a `TestPage` Page, so those are isolated from the setup of logging in?